### PR TITLE
Fixes semver range parsing for required packages

### PR DESCRIPTION
**Pull request by m1kola**
# Description

Moving fix from #413 into a separate PR.

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: internal/catalogmetadata/types.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type Channel struct {
 
 type PackageRequired struct {
 	property.PackageRequired
-	SemverRange *bsemver.Range `json:"-"`
```
Comment (m1kola) [inline internal/catalogmetadata/types.go:34]: I don't think we need a pointer to a function here.

File: internal/catalogmetadata/types.go [hunk 2]
```diff
@@ -98,17 +98,17 @@ func (b *Bundle) loadRequiredPackages() error {
...
 			if err != nil {
 				return fmt.Errorf(
 					"error parsing bundle required package semver range for bundle %q (required package %q): %s",
 					b.Name,
-					requiredPackage.PackageName,
+					requiredPackages[i].PackageName,
 					err,
 				)
 			}
-			requiredPackage.SemverRange = &semverRange
```
Comment (m1kola) [inline internal/catalogmetadata/types.go:111]: So the issue is that we were setting `SemverRange` on a copy and not on the instance from `requiredPackages`.

File: internal/catalogmetadata/types_test.go [hunk 3]
```diff
@@ -123,6 +123,16 @@ func TestBundleRequiredPackages(t *testing.T) {
 	} {
 		t.Run(tt.name, func(t *testing.T) {
 			packages, err := tt.bundle.RequiredPackages()
+			assert.Len(t, tt.wantRequiredPackages, len(packages))
```
Comment (m1kola) [inline internal/catalogmetadata/types_test.go:126]:
```suggestion
			assert.Len(t, packages, len(tt.wantRequiredPackages))
```

**Review by perdasilva — APPROVED** [review]
lgtm

**Pull request merged**